tests/tensorflow/test_tensorflow_estimator_export.py

```diff
@@ -0,0 +1,273 @@
+    with pytest.raises(TypeError, match="Only dict and DataFrame input types are supported"):
+        results = pyfunc_wrapper.predict(inp_list)
+
+
+def test_categorical_model_can_be_loaded_and_evaluated_as_pyfunc(tmpdir, model_path):
+    model_data_info, _ = save_or_log_tf_model_by_mlflow128(
+        str(tmpdir), "categorical", "save_model", model_path
+    )
+
+    pyfunc_wrapper = pyfunc.load_model(model_path)
+
+    # can call predict with a df
+    results_df = pyfunc_wrapper.predict(model_data_info.inference_df)
+    # Precision is less accurate for the categorical model when we load back the saved model.
+    pandas.testing.assert_frame_equal(
+        results_df, model_data_info.expected_results_df, check_less_precise=3
+    )
+
+    # can also call predict with a dict
+    inp_dict = {}
+    for df_col_name in list(model_data_info.inference_df):
+        inp_dict[df_col_name] = model_data_info.inference_df[df_col_name].values
+    results = pyfunc_wrapper.predict(inp_dict)
+    assert isinstance(results, dict)
+    pandas.testing.assert_frame_equal(
+        pandas.DataFrame.from_dict(data=results),
+        model_data_info.expected_results_df,
+        check_less_precise=3,
+    )
+
+    # can not call predict with a list
+    inp_list = []
+    for df_col_name in list(model_data_info.inference_df):
+        inp_list.append(model_data_info.inference_df[df_col_name].values)
+    with pytest.raises(TypeError, match="Only dict and DataFrame input types are supported"):
+        results = pyfunc_wrapper.predict(inp_list)
+
+
+def test_pyfunc_serve_and_score(tmpdir):
+    model_data_info, tracking_uri = save_or_log_tf_model_by_mlflow128(
+        str(tmpdir), "iris", "log_model"
+    )
+    model_uri = f"runs:/{model_data_info.run_id}/model"
+
+    with _use_tracking_uri(tracking_uri):
+        resp = pyfunc_serve_and_score_model(
+            model_uri=model_uri,
+            data=model_data_info.inference_df,
+            content_type=pyfunc_scoring_server.CONTENT_TYPE_JSON_SPLIT_ORIENTED,
+            extra_args=["--env-manager", "local"],
+        )
+    actual = pd.DataFrame(json.loads(resp.content))["class_ids"].values
+    expected = (
+        model_data_info.expected_results_df["predictions"].map(iris_data_utils.SPECIES.index).values
+    )
+    np.testing.assert_array_almost_equal(actual, expected)
+
+
+def test_tf_saved_model_model_with_tf_keras_api(tmpdir):
+    model_data_info, tracking_uri = save_or_log_keras_model_by_mlflow128(
+        str(tmpdir), task_type="log_model", save_as_type="tf1-estimator"
+    )
+
+    def load_and_predict():
+        model_uri = f"runs:/{model_data_info.run_id}/model"
+        with _use_tracking_uri(tracking_uri):
+            mlflow_model = mlflow.pyfunc.load_model(model_uri)
+        predictions = mlflow_model.predict({"features": model_data_info.inference_df})
+        np.testing.assert_allclose(predictions["dense"], model_data_info.expected_results_df)
+
+    load_and_predict()
```

Comment:
```suggestion
    model_uri = f"runs:/{model_data_info.run_id}/model"
    with _use_tracking_uri(tracking_uri):
        mlflow_model = mlflow.pyfunc.load_model(model_uri)
    predictions = mlflow_model.predict({"features": model_data_info.inference_df})
    np.testing.assert_allclose(predictions["dense"], model_data_info.expected_results_df)
```

Do we need a function here?